exercises/practice/palindrome-products/.articles/is-palindrome-check/content.md

```diff
@@ -0,0 +1,165 @@
+# Is number a palindrome? 
+
+There are multiple approaches to solve the palindrome product exercise. 
+Most of them require us to check if a given number is a palindrome. 
```

Comment:
Don't all of them require this?

Reply:
Not all of them. It is possible to generate palindrome numbers in which case you don't need to test for the palindrome. Once you generate the numbers (by composing digits following set patterns) you only have to discover the factors.

Reply:
Ooh neat!

Reply:
This could be an interesting follow up article. The solution becomes more complex than the problem itself so I don't think including a generative approach in the approaches will be helpful, but it might be an interesting thing for those who are into digging deep. 

It would create a nice series of challenges - something I proposed in forum but is not really the way exercises work right now. You could have three connected exercises: Check if a number is a palindrome, palindrome product and refactoring to a generative approach. Each next one could have previous code as a starting point. It's a longer discussion for another day

Reply:
> This could be an interesting follow up article. The solution becomes more complex than the problem itself so I don't think including a generative approach in the approaches will be helpful, but it might be an interesting thing for those who are into digging deep.

Well, it might be actuall! For the reverse string exercise, I also included a Span-based approach which you normally wouldn't use, but could still be interesting to students. I think this would fit well too. But an article would be fine too.